picasso/src/main/java/com/squareup/picasso/RequestCreator.java

```diff
@@ -373,8 +378,19 @@ public void fetch() {
 
       Request request = createRequest(started);
       String key = createKey(request, new StringBuilder());
+      Bitmap bitmap = picasso.quickMemoryCacheCheck(key);
+
+      if (bitmap != null) {
 
-      Action action = new FetchAction(picasso, request, skipMemoryCache, key, tag);
+        if (picasso.loggingEnabled) {
+          log(OWNER_MAIN, VERB_COMPLETED, request.plainId(), "from " + MEMORY);
+        }
+        if (callback != null) {
+          callback.onSuccess();
+        }
+        return;
```

Comment:
I don't like breaking control flow so far down in the method. Returns are fine for early exit but at this point just use an `else`.